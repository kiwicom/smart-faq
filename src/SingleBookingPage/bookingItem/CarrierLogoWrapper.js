// @flow

import * as React from 'react';
import idx from 'idx';
import { CarrierLogo } from '@kiwicom/orbit-components';

import type { Leg } from '../../types';

type Props = {
  legs: Array<Leg>,
};

const CarrierLogoWrapper = (props: Props) => {
  const carriers = {};
  const finalCarriers = [];
  (props.legs || []).forEach(leg => {
    const code = idx(leg, _ => _.airline.code);
    const name = idx(leg, _ => _.airline.name);
    if (code && name && !carriers[code]) {
      carriers[code] = true;
      finalCarriers.push({ code, name });
    }
  });

  return <CarrierLogo size="large" carriers={finalCarriers} />;
};

export const RawCarrierLogoWrapper = CarrierLogoWrapper;

export default CarrierLogoWrapper;
