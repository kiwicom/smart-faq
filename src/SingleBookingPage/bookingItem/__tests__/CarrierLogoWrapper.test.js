// @flow

import * as React from 'react';
import { shallow, render } from 'enzyme';
import ThemeProvider from '@kiwicom/orbit-components/lib/Theming/ThemeProvider';

import { RawCarrierLogoWrapper } from '../CarrierLogoWrapper';

const mockRefType: any = null;
const legs = [
  {
    airline: {
      code: 'PC',
      name: 'Pegasus',
    },
    $refType: mockRefType,
  },
  {
    airline: {
      code: 'PC',
      name: 'Pegasus',
    },
    $refType: mockRefType,
  },
  {
    airline: {
      code: 'FR',
      name: 'Ryanair',
    },
    $refType: mockRefType,
  },
  {
    airline: {
      code: 'FR',
      name: 'Ryanair',
    },
    $refType: mockRefType,
  },
  {
    airline: null,
    $refType: mockRefType,
  },
];

describe('CarrierLogoWrapper', () => {
  it('should render', () => {
    expect(shallow(<RawCarrierLogoWrapper legs={legs} />)).toMatchSnapshot();
  });

  it('should render logo with two carriers', () => {
    const wrapper = render(
      <ThemeProvider>
        <RawCarrierLogoWrapper legs={legs} />
      </ThemeProvider>,
    );

    expect(wrapper.find('img')).toHaveLength(2);
  });
});
