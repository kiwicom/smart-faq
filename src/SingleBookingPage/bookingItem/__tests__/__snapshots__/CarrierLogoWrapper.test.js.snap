// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarrierLogoWrapper should render 1`] = `
<WithTheme(CarrierLogo)
  carriers={
    Array [
      Object {
        "code": "PC",
        "name": "Pegasus",
      },
      Object {
        "code": "FR",
        "name": "Ryanair",
      },
    ]
  }
  size="large"
/>
`;
